connector/src/main/scala/com/google/cloud/spark/bigquery/ArrowBinaryIterator.scala

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2018 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.spark.bigquery
+
+import java.io.{ByteArrayInputStream, InputStream}
+
+import com.google.cloud.spark.bigquery.converters.ArrowSchemaConverter
+import com.google.protobuf.ByteString
+import org.apache.arrow.vector.VectorSchemaRoot
+import org.apache.arrow.vector.ipc.{ArrowReader, ArrowStreamReader}
+import org.apache.spark.sql.catalyst.InternalRow
+import org.apache.spark.sql.execution.arrow.ArrowUtils
+import org.apache.spark.sql.vectorized.{ColumnVector, ColumnarBatch}
+
+import collection.JavaConverters._
+
+/**
+ * An iterator for scanning over rows serialized in Arrow format
+ * @param columnsInOrder Sequence of columns in the schema
+ * @param schema Schema in avro format
+ * @param rowsInBytes Rows serialized in binary format for Arrow
+ */
+class ArrowBinaryIterator(columnsInOrder: Seq[String],
+                          schema: ByteString,
+                          rowsInBytes: ByteString) extends Iterator[InternalRow] {
+
+
+  // max allocation value for the allocator
+  var maxAllocation = Long.MaxValue
+  val allocator = ArrowUtils.rootAllocator.newChildAllocator("ArrowBinaryIterator",
+    0, maxAllocation)
+
+  val byteStringWithSchema = schema.concat(rowsInBytes)
```

Comment:
we should avoid doing the concatenation here at all.

Use either: [SequenceInputStream](https://docs.oracle.com/javase/7/docs/api/java/io/SequenceInputStream.html) or
the pattern in in the [storage example](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/master/bigquery/bigquerystorage/src/main/java/com/example/bigquerystorage/StorageArrowSample.java#L83). 

 Not for this PR but it would be good to verify that creating new iterators for each rowsInBytes isn't a source of overhead.